src/related-items-and-comparisons/OutfitList.jsx

```diff
@@ -5,27 +5,34 @@ const OutfitList = (props) => {
   // hook for array of outfits, starts with nothing by default
   // items must be unique, so clicking does nothing if product is already in the list
   // should always have an "add to list" card
-  const [outfitIDs, updateOutfitIDs] = useState([]);
-  const [outfit, updateOutfit] = useState([]);
-  const [currList, setCurrList] = useState(outfit);
+  var storedFits = JSON.parse(localStorage.getItem('outfit')) || [];
+  var storedIDs = JSON.parse(localStorage.getItem('outfitIDs')) || [];
+  const [outfitIDs, updateOutfitIDs] = useState(storedIDs);
+  const [outfit, updateOutfit] = useState(storedFits);
   const [currIndex, setCurrIndex] = useState(0);
 
+  // using this for a homemade forceUpdate
+  const [val, setVal] = useState(0);
+
+
   const next = () => {
     setCurrIndex(currIndex + 1); // this is changing what currIndex will be the next time the component renders
-    setCurrList(outfit.slice(currIndex + 1));
   };
 
   const prev = () => {
     setCurrIndex(currIndex - 1);
-    setCurrList(outfit.slice(currIndex - 1));
   };
 
-  const addToOutfit = () => { // need to replace with outfit from the current page
+  const addToOutfit = () => {
     if (!outfitIDs.includes(props.currentProduct.product.id)) {
-      // updateOutfit([...outfit, props.products[0].product]);
-      updateOutfit([...outfit, props.currentProduct]);
-      updateOutfitIDs([...outfitIDs, props.currentProduct.product.id]);
+      // updateOutfit(prevState => [...outfit, props.currentProduct]);
+      // updateOutfitIDs([...outfitIDs, props.currentProduct.product.id]);
+      outfit.push(props.currentProduct);
+      outfitIDs.push(props.currentProduct.product.id);
+      setVal(val => val + 1);
```

Comment:
Once you fix your state management, `val` and `setVal` will no longer be needed, which is good, because they shouldn't be there in the first place.

Reply:
fixed